CorsixTH/Lua/entities/humanoid.lua

```diff
@@ -841,6 +852,29 @@ function Humanoid:unregisterRoomRemoveCallback(callback)
   end
 end
 
+
+-- Registers a new staff callback for this humanoid.
+--!param callback (function) The callback to call when a staff member has been hired or fired
+function Humanoid:registerStaffCallback(callback)
+  if self.staff_callbacks and not self.staff_callbacks[callback] then
+    self.staff_callbacks[callback] = true
```

Comment:
I just copied the existing callback related functions that do the same. To me in theory, it looks like we only should ever have one call back at a time registered for each type, but I haven't checked that specifically but the individual action callbacks are unregistered, so its when each callback is satisfied they should unregister themselves, but then you only 'seek one room at a time' - ward and operating theatre being special for treatment. Difference here is that for diagnosis rooms you might have a list of possible diagnosis rooms to try, and you'd get another callback for another SeekRoomAction but I'd have to check that too.

Maybe it was envisioned as having a list of actions that were toggled for activity for the life of the patient rather than recreating them.

Though I can see in issue #1337 for example, where a callback for a GPs office exists, when GPs offices already exist. In that particular issue, many patients get redirected to the newly built room, as the build callbacks exist, whereas room:tryToFindNearbyPatients should probably be called instead or the callback not exist at all.